packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,224 @@
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=13227%3A163026) to all variants of the Checkbox component
+
+## `Checkbox` API
+
+| Prop                | Type       | Required | Default   | Description                                                                                                                                              |
+| ------------------- | ---------- | -------- | --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| children            | `string`   | No       | undefined | The text to be rendered as CheckboxLabel                                                                                                                 |
+| isChecked           | `boolean`  | No       | false     | state of the checkbox, if explicitly set the checkbox will become [controlled component](https://reactjs.org/docs/forms.html#controlled-components)                                                                                 |
+| defaultChecked      | `boolean`  | No       | false     | default state of the checkbox, if isChecked is not provided the checkbox will become [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html)                                                            |
+| onChange            | `({ event, isChecked, value }) => void` | No       | undefined | The function to be called when the checkbox state changes.                                                                                               |
+| isIndeterminate     | `boolean`  | No       | false     | mixed state of the checkbox                                                                                                                              |
+| isDisabled          | `boolean`  | No       | `false`   | Control whether the checkbox is disabled or not.                                                                                                         |
+| isRequired          | `boolean`  | No       | `false`   | Control whether the checkbox is required or not.                                                                                                         |
+| helpText          | `string`   | No       | undefined | The helper text to be rendered                                                                                                                           |
+| errorText          | `string`   | No       | undefined | The error text to be rendered                                                                                                                           |
+| validationState           | `undefined \| 'error'`  | No       | `undefined`   | Control whether the checkbox is invalid or not.                                                                                                          |
```

Comment:
do we really need `undefined` as a value?

Reply:
Yes we should add undefined as value. Since we are treating it as a value. 

For example in code also we should do this: 

```ts
// explicitly setting undefined in the union
type CheckboxProps = {
  validationState?: 'error' | undefined;
}
```

**But Why?** 

because if consumers set [exactOptionalPropertyTypes](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) in their tsconfig.json they will get error that undefined is not assignable. 

```tsx
<Checkbox validationState={undefined}>Hello</Checkbox>;
^
Type '{ children: string; validationState: undefined; }' is not assignable to type 'CheckboxProps' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
  Types of property 'validationState' are incompatible.
    Type 'undefined' is not assignable to type '"error"'.ts(2375)
```

**[More info in this Slack Thread ](https://razorpay.slack.com/archives/C02FRV92MPH/p1651860634820809?thread_ts=1651851415.416079&cid=C02FRV92MPH)**

Reply:
okay then if that's the case i would recommend to have a semantic value for undefined state(and we can use this everywhere then even in text input I'll update).
basically,

```
validationState = 'error' | 'success' | 'none'
```

that would be better to write in code too

```
validationState={text.length < 10 ? 'error' : 'none'}
```

Reply:
Yes that's what I'm also pushing for https://github.com/razorpay/blade/pull/552#discussion_r926472221 

'none' seems fine. cc @nashcheez